src/main/java/reposense/authorship/FileInfoAnalyzer.java

```diff
@@ -62,19 +66,68 @@ public static FileResult analyzeFile(RepoConfiguration config, FileInfo fileInfo
             return null;
         }
 
-        return generateFileResult(fileInfo);
+        return generateTextFileResult(fileInfo);
+    }
+
+    /**
+     * Analyzes the binary file, given in the {@code fileInfo}, that has changed in the time period provided
+     * by {@code config}.
+     * Returns null if the file is missing from the local system, or none of the
+     * {@code Author} specified in {@code config} contributed to the file in {@code fileInfo}.
+     */
+    public static FileResult analyzeBinaryFile(RepoConfiguration config, FileInfo fileInfo) {
+        String relativePath = fileInfo.getPath();
+
+        if (Files.notExists(Paths.get(config.getRepoRoot(), relativePath))) {
+            logger.severe(String.format(MESSAGE_FILE_MISSING, relativePath));
+            return null;
+        }
```

Comment:
Can add a test that ends up displaying this image? Need not be an elaborate one.

Reply:
Sure, no problem